sysutils/pfSense-pkg-syslog-ng/files/usr/local/www/syslog-ng_log_viewer.php

```diff
@@ -54,118 +54,111 @@
+		$grepcmd = "$grep -v " . escapeshellarg($filter) . " $logfile";
 	} else {
-		$grepcmd = "$grep '$filter' $logfile";
+		$grepcmd = "$grep  " . escapeshellarg($filter) . " $logfile";
 	}
 
 	if ($archives) {
 		$grepcmd = $grepcmd . "*";
 	}
 
-	$log_lines = trim(shell_exec("$grepcmd | wc -l"));
-	$log_output = trim(shell_exec("$grepcmd | sort -M | tail -n $limit"));
+	$log_lines = trim(shell_exec("$grepcmd | /usr/bin/wc -l"));
+	$log_output = trim(shell_exec("$grepcmd | /usr/bin/sort -M | /usr/bin/tail -n $limit"));
 
 	if (!empty($log_output)) {
 		$log_messages = explode("\n", $log_output);
 		$log_messages_count = sizeof($log_messages);
 	}
 }
 
-$pgtitle = array(gettext("Package"), gettext("Services: Syslog-ng"), gettext("Logs"));
-
-include("head.inc");
-?>
-
-<?php include("fbegin.inc"); ?>
-<?php if ($savemsg) print_info_box($savemsg); ?>
-
-<style>
-
-.border-bottom {
-	border: 1px solid #F5F5F5; border-width: 0px 0 3px 0px;
+$pgtitle = array("Package", "Services: Syslog-ng", "Logs");
+require_once("head.inc");
+if ($savemsg) {
+	print_info_box($savemsg);
 }
-
-</style>
-
-
-<form action="syslog-ng_log_viewer.php" method="post" name="iform">
-<table width="100%" border="0" cellpadding="0" cellspacing="0" style="background-color: #F5F5F5;">
-	<tr><td>
-<?php
-	$tab_array = array();
-	$tab_array[] = array("General", false, "/pkg_edit.php?xml=syslog-ng.xml&amp;id=0");
-	$tab_array[] = array("Advanced", false, "/pkg.php?xml=syslog-ng_advanced.xml");
-	$tab_array[] = array("Log Viewer", true, "/syslog-ng_log_viewer.php");
-	display_top_tabs($tab_array);
+$tab_array = array();
+$tab_array[] = array("General", false, "/pkg_edit.php?xml=syslog-ng.xml&amp;id=0");
+$tab_array[] = array("Advanced", false, "/pkg.php?xml=syslog-ng_advanced.xml");
+$tab_array[] = array("Log Viewer", true, "/syslog-ng_log_viewer.php");
+display_top_tabs($tab_array);
 ?>
-	</td></tr>
-	<tr><td>
-	<div id="mainarea">
-		<table id="maintable" class="tabcont" width="100%" border="0" cellpadding="0" cellspacing="0">
-			<tr><td>
 
-			<table width="100%" class="panel-default" style="background-color: #FFFFFF;">
-				<h2 class="panel-title" style="background-color: #424242; color: #FFFFFF; border: solid 5px #424242;">Syslog-ng Logs</h2>
-				<tr><td class="border-bottom" width="22%">Log File</td><td class="border-bottom" width="78%"><select name="logfile">
-				<?php
-				$log_files = syslogng_get_log_files($objects);
-				foreach($log_files as $log_file) {
-					if($log_file == $logfile) {
-						echo "<option value=\"$log_file\" selected=\"selected\">$log_file</option>\n";
-					} else {
-						echo "<option value=\"$log_file\">$log_file</option>\n";
+<form action="syslog-ng_log_viewer.php" method="post" name="iform">
+<div class="panel panel-default">
+	<div class="panel-heading"><h2 class="panel-title"><?=gettext("Syslog-ng Log Viewer Settings")?></h2></div>
+	<div class="panel-body table-responsive">
+	<table class="table table-condensed">
+		<tbody><tr><td>
```

Comment:
This code should be converted to PHP Form classes for consistent appearance and to avoid nested tables. e.g.:

```
$section = new section("Syslog-ng Log Viewer Settings");

$section->addInput(Form_Select(
   'logfile',
   'Log File',
   ...
```